packages/components/src/tabs/use-tabs.ts

```diff
@@ -433,33 +404,33 @@ export function useTabIndicator(): React.CSSProperties {
 
   // Update the selected tab rect when the selectedIndex changes
   useSafeLayoutEffect(() => {
-    if (selectedIndex == null) return
+    if (selectedValue == null) return
 
-    const tab = descendants.item(selectedIndex)
+    const tab = document.getElementById(`${makeTabId(id, selectedValue)}`)
```

Comment:
We want to make sure this is scoped to the correct owner document.

Can we do something like `ref?.current.ownerDocument`

Reply:
I was using querySelector initially on the root element, but that somehow would throw invalid selector every now and then. Your suggestion would work nicely.